### PR TITLE
Menu component for UI

### PR DESCRIPTION
**Pull request by Noorae**
### New additions:

MenuDrawer component
- Drawer is used for screens wider than 600px by Material UI's breakpoints
- SwipeableDrawer is used for small screens (mobile)
- Known issue: opening button functionality needs to be added for small desktop screen (smaller than 600px) to open the drawer as it is currently opened only by mobile touch effects and cannot be clicked open. 
- Known issue: the menu goes on top of the slider
- Known issue: Map drags when menu is opened on mobile with swipe

DatePickers component
- Correct dates can be used with the datepicker component now, and these dates are used with DateSlider component.
- Still need to add functionality to open satellite view with the DatePicker components, currently only sets the dates.
- Known issue: sometimes the dates on the slider do not update correctly. 

custom LayerControl component
- Functionality still missing, currently template with mock data

Other
- added Materia UI Date Pickers as a dependency

### Changes:

- Changed the DateSlider to use correct startDate and endDate instead of Period
- Search bar gives space to menu drawer on desktop view

### CODE REVIEW

File: app/components/controls/Controls.tsx [hunk 1]
```diff
@@ -12,24 +19,31 @@ export interface Period {
 
 export interface ControlsProps {
   period: Period;
```
[the remark this answers is not in the document]
Comment (Noorae) [reply app/components/controls/Controls.tsx]: done

File: app/components/map/_MapComponent.client.tsx [hunk 2]
```diff
@@ -119,6 +119,8 @@ export default function MapComponent() {
           setStartDate={setStartDate}
```
Comment (kuukkanen) [inline app/components/map/_MapComponent.client.tsx:119]: Remember to remove `period` from here as well 👆🏻

**Review by kuukkanen — CHANGES_REQUESTED** [review]

File: app/components/map/_MapComponent.client.tsx (hunk 2 above)
Comment (Noorae) [reply app/components/map/_MapComponent.client.tsx:118]: done! Hopefully I didn't miss it elsewhere.

**Review by kuukkanen — APPROVED** [review]
LGTM.

File: app/components/map/_MapComponent.client.tsx (hunk 2 above)
Comment (kuukkanen) [reply app/components/map/_MapComponent.client.tsx:118]: You kind of did, but that's fine.

**Pull request merged**